From 0e773ed899d01b708aeaaafcad291a3893da2ba1 Mon Sep 17 00:00:00 2001
From: Martin Finkel <martin@videolabs.io>
Date: Mon, 15 Jan 2018 11:41:31 +0100
Subject: [PATCH] Revert "contrib: gnutls: update to current stable 3.5.16
 (fixes #19350)"

This reverts commit 496d5f426d7fde1bb60c90a0009d222a52fd582d.

# Conflicts:
#	contrib/src/gnutls/rules.mak
---
 contrib/src/gnutls/SHA512SUMS                    | 2 +-
 contrib/src/gnutls/gnutls-pkgconfig-static.patch | 9 ++++-----
 contrib/src/gnutls/rules.mak                     | 5 ++---
 3 files changed, 7 insertions(+), 9 deletions(-)

diff --git a/contrib/src/gnutls/SHA512SUMS b/contrib/src/gnutls/SHA512SUMS
index e5e3af8db4..7292832e42 100644
--- a/contrib/src/gnutls/SHA512SUMS
+++ b/contrib/src/gnutls/SHA512SUMS
@@ -1 +1 @@
-451d3167be599ed8e0333dd7c9f8501fcb47b7aa871aeb461c368381c0b7ecd7e2026ec35dbbb2aa685cb2c3a22e9296e0a0699409e3744b731c1bb7e7e69f07  gnutls-3.5.16.tar.xz
+d75f6b4dea2dc742cd7f60ee0ee540d41b69991aaa937ca0138cfdf4a1e0dfaaa3863464303bfa5799e14ee02de252f71c59a7a9e57b96ff8af653e419edfd4e  gnutls-3.4.14.tar.xz
diff --git a/contrib/src/gnutls/gnutls-pkgconfig-static.patch b/contrib/src/gnutls/gnutls-pkgconfig-static.patch
index 0a5ecfcef0..361d65104b 100644
--- a/contrib/src/gnutls/gnutls-pkgconfig-static.patch
+++ b/contrib/src/gnutls/gnutls-pkgconfig-static.patch
@@ -1,11 +1,10 @@
-diff -ru gnutls-3.5.16/lib/gnutls.pc.in gnutls-3.5.16.patched/lib/gnutls.pc.in
---- gnutls-3.5.16/lib/gnutls.pc.in	2017-03-28 20:20:50.000000000 +0300
-+++ gnutls-3.5.16.patched/lib/gnutls.pc.in	2017-12-26 16:47:54.502835713 +0200
+--- gnutls/lib/gnutls.pc.in    2015-11-03 21:47:28.000000000 +0000
++++ gnutls.patched/lib/gnutls.pc.in    2016-11-02 22:53:04.272156182 +0000
 @@ -19,6 +19,6 @@
  URL: http://www.gnutls.org/
  Version: @VERSION@
  Libs: -L${libdir} -lgnutls
--Libs.private: @LIBZ_PC@ @LIBINTL@ @LIBSOCKET@ @LIBNSL@ @LIBPTHREAD@ @LIB_SELECT@ @TSS_LIBS@ @GMP_LIBS@ @LIBUNISTRING@ @LIBIDN2_LIBS@
-+Libs.private: @LIBZ@ @LIBINTL@ @LIBSOCKET@ @LIBNSL@ @LIBPTHREAD@ @LIB_SELECT@ @TSS_LIBS@ @GMP_LIBS@ @LIBUNISTRING@ @LIBIDN2_LIBS@ @LIBIDN_LIBS@ @HOGWEED_LIBS@ @NETTLE_LIBS@
+-Libs.private: @LIBZ@ @LIBINTL@ @LIBSOCKET@ @LIBPTHREAD@ @LIBICONV@ @P11_KIT_LIBS@ @LIB_SELECT@ @TSS_LIBS@ @GMP_LIBS@
++Libs.private: @LIBZ@ @LIBINTL@ @LIBSOCKET@ @LIBPTHREAD@ @LIBICONV@ @P11_KIT_LIBS@ @LIB_SELECT@ @TSS_LIBS@ @GMP_LIBS@ @NETTLE_LIBS@ @LIBIDN_LIBS@ @HOGWEED_LIBS@
  @GNUTLS_REQUIRES_PRIVATE@
  Cflags: -I${includedir}
diff --git a/contrib/src/gnutls/rules.mak b/contrib/src/gnutls/rules.mak
index 9cd29beead..66a3b00d64 100644
--- a/contrib/src/gnutls/rules.mak
+++ b/contrib/src/gnutls/rules.mak
@@ -1,7 +1,7 @@
 # GnuTLS
 
-GNUTLS_VERSION := 3.5.16
-GNUTLS_URL := ftp://ftp.gnutls.org/gcrypt/gnutls/v3.5/gnutls-$(GNUTLS_VERSION).tar.xz
+GNUTLS_VERSION := 3.4.14
+GNUTLS_URL := ftp://ftp.gnutls.org/gcrypt/gnutls/v3.4/gnutls-$(GNUTLS_VERSION).tar.xz
 
 ifdef BUILD_NETWORK
 ifndef HAVE_DARWIN_OS
@@ -57,7 +57,6 @@ GNUTLS_CONF := \
 	--disable-doc \
 	--disable-tests \
 	--with-included-libtasn1 \
-	--with-included-unistring \
 	$(HOSTCONF)
 
 GNUTLS_ENV := $(HOSTVARS)
-- 
2.15.1.windows.2

